src/renderer/services/wallet/balances.ts

```diff
@@ -34,29 +35,107 @@ export const reloadBalances: FP.Lazy<void> = () => {
   BCH.reloadBalances()
 }
 
-export const reloadBalancesByChain: (chain: Chain) => FP.Lazy<void> = (chain) => {
+export const reloadBalancesByChain: (chain: Chain) => (state?: 'trigger' | '') => void = (chain) => {
   switch (chain) {
     case 'BNB':
-      return BNB.reloadBalances
+      return (state) => BNB.reloadBalances(state)
     case 'BTC':
-      return BTC.reloadBalances
+      return (state) => BTC.reloadBalances(state)
     case 'BCH':
-      return BCH.reloadBalances
+      return (state) => BCH.reloadBalances(state)
     case 'ETH':
-      return ETH.reloadBalances
+      return (state) => ETH.reloadBalances(state)
     case 'THOR':
-      return THOR.reloadBalances
+      return (state) => THOR.reloadBalances(state)
     case 'LTC':
-      return LTC.reloadBalances
+      return (state) => LTC.reloadBalances(state)
     default:
       return () => {}
   }
 }
 
+let balanceRecord: Partial<Record<Chain, WalletBalancesRD>> = {}
+
+export const clearSavedBalances = () => {
+  balanceRecord = {}
+}
+
+const getChainBalanceData$: (chain: Chain) => WalletBalancesLD = (chain) => {
+  switch (chain) {
+    case 'BNB':
+      return BNB.balances$
+    case 'BTC':
+      return BTC.balances$
+    case 'BCH':
+      return BCH.balances$
+    case 'ETH':
+      return FP.pipe(
+        network$,
+        RxOp.switchMap((network) => ETH.balances$(network === 'testnet' ? ETHAssets : undefined))
+      )
+    case 'THOR':
+      return THOR.balances$
+    case 'LTC':
+      return LTC.balances$
+    default:
+      return Rx.of(RD.initial)
+  }
+}
+
+const getChainBalanceReload$ = (chain: Chain) => {
+  switch (chain) {
+    case 'BNB':
+      return BNB.reloadBalances$
+    case 'BTC':
+      return BTC.reloadBalances$
+    case 'BCH':
+      return BCH.reloadBalances$
+    case 'ETH':
+      return ETH.reloadBalances$
+    case 'THOR':
+      return THOR.reloadBalances$
+    case 'LTC':
+      return LTC.reloadBalances$
+    default:
+      return Rx.EMPTY
+  }
+}
+
+const getChainBalance$ = (chain: Chain): WalletBalancesLD => {
+  const reload$ = FP.pipe(
+    getChainBalanceReload$(chain),
+    RxOp.finalize(() => {
```

Comment:
[question] In which case can we expect this stream will be completed?

Reply:
as i can see it (and adding some `console.log` expression inside of `finalize` confirms it) as we have `triggerStream` as a `BehaviourSubject` it will be completed on unsubscribe and in all of our cases unsubscribe will be initiated on any View unmount (comment for that is already added)